plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -354,7 +354,7 @@ open class DefaultPageCreator(
             val extensions = (classlikes as List<WithExtraProperties<DClasslike>>).flatMap {
                 it.extra[CallableExtensions]?.extensions
                     ?.filterIsInstance<Documentable>().orEmpty()
-            }
+            }.distinctBy{ it.sourceSets to it.dri} // comparing of [Documentable] is expensive
```

Comment:
Very good idea to leave a comment! I'd stress that it's expensive at the time of writing this code, so that we or other maintainers are not afraid to change it in the future. 

Hopefully, one day `Documentable` will be migrated from a data class to an average class, or at the very least have an inexpensive equals/hashcode implemented

```suggestion
            }.distinctBy { it.sourceSets to it.dri} // [Documentable] has expensive equals/hashCode at the moment, see #2620
```